### PR TITLE
tooling: Add dev requirements setup

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:

This allows requirements without hashes - ie using `-e` to be used in development

They can either point to a git/hub requirement for testing in CI, or a local filesystem requirement for development of pytooling packages.

Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @kfaseela @htuch @moderation

File: tools/dev/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Remote-pdb can be used inside build jobs, or parallelized aspects
+remote-pdb
+
+# Add temporary dev requirements with `-e` here
+#
+# Note: This should always be empty on `main`.
```
Comment (htuch) [inline tools/dev/requirements.txt:6]: Add a CI check for this?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -26,6 +26,11 @@ updates:
   schedule:
     interval: "daily"
 
+- package-ecosystem: "pip"
```
Comment (htuch) [inline .github/dependabot.yml:29]: Why is this needed if always empty on main?

File: tools/dev/requirements.txt (hunk 1 above)
Comment (phlax) [reply tools/dev/requirements.txt:6]: i thought about this, its not trivial to do without making this useless for ci

ie ci fails the deps check as it has a dev requirement, but then doesnt use the dev requirement that you want to test

File: .github/dependabot.yml (hunk 2 above)
Comment (phlax) [reply .github/dependabot.yml:29]: i have to ~permanently add this to work on any pytooling or external python libs, so it would be v helpful for me not having to add this boilerplate every time

its come up now as im showing @kfaseela how pytooling works and can be worked on with Envoy main, and working on some dev docs

adding this will make documenting dev/ci testing with pytooling a lot easier

File: tools/dev/requirements.txt (hunk 1 above)
Comment (htuch) [reply tools/dev/requirements.txt:6]: Different CI stages?

Comment (phlax) [reply tools/dev/requirements.txt:6]: that works as long as the its not used by any later stage - currently the check would go in the dep check which would prevent anything after the first stage

i can add it, but on balance i would prefer to keep this as a visual inspection in PRs where it has been (temporarily added)

Comment (htuch) [reply tools/dev/requirements.txt:6]: OK, I don't feel that strongly, just generally prefer machines do the work.

File: .github/dependabot.yml (hunk 2 above)
Comment (phlax) [reply .github/dependabot.yml:29]: rereading this, and im wondering if perhaps you meant why was the dependabot entry required

that is because we have a check for it - we could potentially adjust the check to ignore the dev requirement and remove this

**Review by kfaseela — APPROVED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (htuch) [reply .github/dependabot.yml:29]: Yes, that is what I meant. Might be best to skip the check.

Comment (phlax) [reply .github/dependabot.yml:29]: sure, ill hack the `pip_check` to ignore...

Comment (phlax) [reply .github/dependabot.yml]: pip_check updated and the dependabot stub removed

**Review by htuch — APPROVED** [review]
LGTM, thanks!
